nixos/nextcloud: Update logreader warning description

provokateurin opened the pull request:
The error is no longer logged since https://github.com/nextcloud/logreader/pull/1449. This was a bug in the app, since the error is not fixable by the admin due to the logging configuration.

CC @schneefux @bachp @globin @Ma27 @britter

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

provokateurin committed:
nixos/nextcloud: Update logreader warning description

The error is only logged as an info since https://github.com/nextcloud/logreader/pull/1449.
This was a bug in the app, since the error is not fixable by the admin due to the logging configuration.

GetPsyched commented on nixos/doc/manual/redirects.json:
```diff
@@ -524,9 +524,6 @@
   "module-services-nextcloud-known-warnings": [
     "index.html#module-services-nextcloud-known-warnings"
   ],
-  "module-services-nextcloud-warning-logreader": [
-    "index.html#module-services-nextcloud-warning-logreader"
-  ],
```
Could you instead prepend an entry here that points to a section in the release notes stating why this was removed?
cc @fricklerhandwerk

provokateurin replied:
I can try, but I'm not sure how to properly do that, also given the mentioned PR which seems to restructure the changelogs.

SuperSandro2000 replied:
What should we write there? "It is no longer relevant". I'm having my fair issue thinking this would help anyone.

fricklerhandwerk replied:
We could note that the log reader application was fixed, or whatever the reason is for this note to get removed.

provokateurin commented:
Alternatively we can also keep the section, as the message is still shown, but only as info and not as an error anymore. Having the section might still be useful for some people to understand why this message appears.

provokateurin commented:
I simply changed the description to match the new behavior, as the "warning" is still shown and admins might want to understand what it means.

Ma27 commented:
Oof sorry this completely slipped. Taking a look...

Ma27 commented on nixos/modules/services/web-apps/nextcloud.md:
```diff
@@ -245,18 +245,18 @@ that are managed by Nix. If you want automatic updates it is recommended that yo
 * the Logreader application that allows reading logs in the admin panel is enabled
   by default and requires logs written to a file.
 
-The logreader application doesn't work, as it was the case before. The only change is that
```
I think this sentence doesn't fit here anymore, does it?

provokateurin replied:
I'm not sure what you mean? I removed the sentence, so yes it doesn't fit anymore.

Ma27 replied:
If I'm reading the diff right, "The logreader application doesn't work, as it was the case before." is still there. And I don't think it really fits (anymore).

provokateurin replied:
I still don't understand what you want me to change. Can you either explain again in a different way or directly push to this PR if you don't mind?

Ma27 commented:
With the last comment resolved, I think we can merge this.
Adding a backport label since the change got backported to nc 28-30 and thus it's also in 24.11, so we can also update its manual.

Ma27 committed:
nixos/nextcloud: reword docs about logs a bit

Originally, I only wanted to remove
"The logreader application doesn't work, as it was the case before.".

But then, the rest sounded a little weird, so I reworded the paragraph a
bit more ;-)

Ma27 commented:
@provokateurin pushed 94307683c174357137ec99c2ea82b6ebe02c29a3. I originally wanted to remove the first sentence only, but then the rest sounded a little weird.

Let me know if you're OK with that change, then I'd merge!
That said, thanks for the upstream work for that in the logreader app :)

provokateurin commented:
LGTM
